### PR TITLE
fix: 🐛 WIN-132 Getting same price for 1 room and 5 rooms

### PR DESCRIPTION
**Pull request by tomashq**
Price was too low in case of multiple rooms were searched

✅ Closes: [WIN-132]

[WIN-132]: https://simard.atlassian.net/browse/WIN-132?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by tomashq** [comment]
On some pages we need to display price per night and per room, on others it is price per night for all rooms
Introduced flexible way to calculate price (per night, all nights, per room, all rooms and combinations)

File: src/hooks/useCurrencies.ts [hunk 1]
```diff
@@ -88,10 +88,12 @@ export const useCurrencies = () => {
   const convertPriceCurrency = useCallback(
     ({
       price,
-      targetCurrency
+      targetCurrency,
+      amount
     }: {
       price: Price;
       targetCurrency: CurrencyCode;
+      amount: number;
```
Comment (dave-ok) [inline src/hooks/useCurrencies.ts:96]: I am assuming `amount` is supposed to be a multiplier for the Price object to ensure all values are multiplied.
Maybe we could give it a different name, say `multiplier` so it is obvious, and it is not assumed it's the actual monetary amount. 

Just a thought, maybe we could instead, create a helper function say `PriceMapper` or`PriceMultiplier` or whatever... that accepts a function for transforming the monetary fields with for example a multiplification function. We wont need to introduce the multiplier parameter anymore. We could do that outside this function if we need to transform/multiply the price by a multiple

That mapper function could also be used within this converter as well [below](https://github.com/windingtree/win-frontend/blob/99c9b8149b608bc43446f080d86a520a9c1892fa/src/hooks/useCurrencies.ts#L119-L127)

File: src/hooks/useCurrencies.ts [hunk 2]
```diff
@@ -102,7 +104,8 @@ export const useCurrencies = () => {
 
       const convertedCurrency = convertCurrency(
         price.currency as CurrencyCode,
-        targetCurrency
+        targetCurrency,
+        amount
       );
```
Comment (dave-ok) [inline src/hooks/useCurrencies.ts:109]: Seeing as we are returning early here, the `amount` or `multiplier` will not be applied if the source and target currencies are the same

File: src/hooks/useAccommodationsAndOffers/helpers.ts [hunk 3]
```diff
@@ -46,48 +46,47 @@ export const getActiveAccommodations = (
...
+      if (!perRoom) {
+        multiplier *= numberOfRooms;
+      }
+      if (perNight) {
+        divider *= numberOfDays;
       }
+      return {
+        price: (Number(totalPricePerRoom) / divider) * multiplier,
+        currency
+      };
```
Comment (dave-ok) [inline src/hooks/useAccommodationsAndOffers/helpers.ts:89]: This is an awesome idea! The function handles 4 different evaluations depending on input. This is fine as it is.

Looking at it more closely now, as a suggestion, could we go one step further and just return all 4 evaluations in an object everytime, with the fields appropriately named `perRoomAllNights`, `perRoomPerNight`, `perNightAllRooms`, `AllRoomsAllNights`. 

Then it is up to the user to determine what they need. We will no longer need `perRoom` and `perNight` parameters. That way we probably won't mix up the parameters when we need a particular combination. and it is clear at first glance, what value we are using from the return of this function.

File: src/hooks/useCurrencies.ts [hunk 4]
```diff
@@ -102,7 +104,8 @@ export const useCurrencies = () => {
```
Comment (dave-ok) [inline src/hooks/useCurrencies.ts:104]: Seeing as we are returning early here, the `amount` or `multiplier` will not be applied if the source and target currencies are the same

**Review by bobetbat — APPROVED** [review]
LGTM

**Pull request merged**